config/initializers/rpush.rb

```diff
@@ -52,8 +52,15 @@
   # end
 
   # Called when a notification is successfully delivered.
-  # on.notification_delivered do |notification|
-  # end
+  on.notification_delivered do |notification|
+    ForemStatsClient.increment(
+      "push_notifications.delivered",
+      tags: [
+        "app_id:#{notification.app_id}",
+        "bundle_id:#{notification&.app&.bundle_id}",
+      ],
```

Comment:
@fdoxyz Are we using/registering any kind of category for push notifications? I'm concerned that we won't be able to record what type of notification we're sending: that's not an issue now with only one push notification type being send, but I can see it being an issue in the future!

Reply:
I agree 100% that we should have as much detail as possible here! Not only so it will be easier in the future to troubleshoot when necessary, but also having better observability altogether can help us make decisions based on the usage of each notification.

Right now we aren't adding anything to track this. From a quick glance I see two options to try and do this:

1. Add `type: 'Notifications::NewComment::Send'` to the payload [here](https://github.com/forem/forem/blob/master/app/services/notifications/new_comment/send.rb#L52)
   - This is basically adding the name of the service that is responsible of this notification, but we can settle in some other name too
2. Add a `type` attribute (maybe `notification_type` or some other name) to the model itself
   - This would require to [add an extra attribute](https://github.com/rpush/modis#usage) to the [Rpush notification model](https://github.com/rpush/rpush/blob/master/lib/rpush/client/redis/notification.rb). I'm not sure but I think this would require extending or monkeypatching that model (?)
3. Any other ideas?

With either one of the above we can then add the tag here to be sent into `ForemStatsClient`. IMO I think option 1 would be a little easier/cleaner but option 2 _might_ be a bit more robust solution overall. What do you think about these?

Reply:
Option #1 would be my preferred option: adding attributes to the Rpush notification could really set us up for issue down they road if they change their object model. Using the payload field to add our own data seems like a pretty safe way to do this: we're specifically using the area Rpush intends user and notifications specific data to go!

Reply:
Just a small note that we can update the data reported in the tags following the [suggestion here](https://github.com/forem/forem/pull/13118#discussion_r604149408)